options/ansi/generic/stdio.cpp

```diff
@@ -934,7 +934,8 @@ int fputc(int c, FILE *stream) {
 }
 
 int fputs_unlocked(const char *__restrict string, FILE *__restrict stream) {
-	if(fwrite_unlocked(string, strlen(string), 1, stream) != 1)
+	size_t length = strlen(string);
```

Comment:
It took me a while to see it, but the reason why this is functionally different from the previous state is that `fwrite` with a `length` (or nitems) of zero will return 0. It might be nice to explain that in a comment.

Nit: the commit message should start with `options/ansi` instead of `ansi`.

Reply:
My reasoning for starting the commit message with `ansi:` was that it also touched tests, but I understand that the main change in that commit is in options so I'll change that accordingly.